modmail/utils/pagination.py

```diff
@@ -0,0 +1,260 @@
+        if not (is_valid := self.source_message.author.id == interaction.user.id):
+            await interaction.response.send_message(
+                content="This is not your message to paginate!", ephemeral=True
+            )
+        return is_valid
+
+    def modify_states(self) -> None:
+        """Disable specific components depending on paginator page and length."""
+        less_than_2_pages = len(self.pages) <= 2
+        components = {
+            "jump_first": less_than_2_pages,
+            "back": False,
+            "next": False,
+            "jump_last": less_than_2_pages,
+        }
+
+        if self._index == 0:
+            components["jump_first"] = True
+            components["back"] = True
+
+        if self._index == len(self.pages) - 1:
+            components["next"] = True
+            components["jump_last"] = True
+
+        for child in self.children:
+            if child.custom_id in components.keys():
+                if getattr(child, "disabled", None) is not None:
+                    child.disabled = components[child.custom_id]
+                    if getattr(child, "style", None) is not None:
+                        child.style = ButtonStyle.secondary if child.disabled else ButtonStyle.primary
+
+    async def send_page(self, interaction: "Interaction") -> None:
+        """Send new page."""
+        self.modify_states()
+        await interaction.message.edit(content=self.pages[self._index], view=self)
+
+    @button(label=JUMP_FIRST_LABEL, custom_id="jump_first", style=ButtonStyle.primary)
+    async def go_first(self, button: "Button", interaction: "Interaction") -> None:
+        """Move the paginator to the first page."""
+        if self._index == 0:
+            return
+
+        self._index = 0
+        await self.send_page(interaction)
+
+    @button(label=BACK_LABEL, custom_id="back", style=ButtonStyle.primary)
+    async def go_previous(self, button: "Button", interaction: "Interaction") -> None:
+        """Move the paginator to the previous page."""
+        if self._index == 0:
+            return
+
+        self._index -= 1
+        await self.send_page(interaction)
+
+    @button(label=FORWARD_LABEL, custom_id="next", style=ButtonStyle.primary)
+    async def go_next(self, button: "Button", interaction: "Interaction") -> None:
+        """Move the paginator to the next page."""
+        if self._index == len(self.pages) - 1:
+            return
+
+        self._index += 1
+        await self.send_page(interaction)
+
+    @button(label=JUMP_LAST_LABEL, custom_id="jump_last", style=ButtonStyle.primary)
+    async def go_last(self, button: "Button", interaction: "Interaction") -> None:
+        """Move the paginator to the last page."""
+        if self._index == len(self.pages) - 1:
+            return
+
+        self._index = len(self.pages) - 1
+        await self.send_page(interaction)
+
+    @button(emoji=STOP_PAGINATE_EMOJI, custom_id="stop_paginate", style=ButtonStyle.grey)
```

Comment:
Nitpick, should be red?

Reply:
Red emoji on a red button doesn't go well together. If there's a better emoji option that isn't red (built in to discord) I'd love to switch to it.